### PR TITLE
fix: track best validation metric for darts_cnn hp search benchmark

### PR DESCRIPTION
**Pull request by liamcli**
## Description
The validation accuracy is quiet noisy and fluctuates a lot for this benchmark.  This negatively impacts the performance of hp search, which uses the latest validation accuracy to make early-stopping decisions.  Hence, we change the validation accuracy to return the best accuracy so far instead of latest so that the searcher has a higher quality signal for early-stopping.  

## Test Plan
Verified with experiments that adaptive search does indeed find architectures with better final accuracy when using best accuracy so far as the searcher metric.

### CODE REVIEW

File: examples/hp_search_benchmarks/darts_cifar10_pytorch/adaptive.yaml [hunk 1]
```diff
@@ -142,6 +142,7 @@ searcher:
     batches: 156200
   max_trials: 10000
   mode: aggressive
+  max_rungs: 4
```
Comment (naren-determined) [inline examples/hp_search_benchmarks/darts_cifar10_pytorch/adaptive.yaml:145]: Might be helpful to leave an inline comment here with some reasoning.

File: examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py [hunk 2]
```diff
@@ -149,7 +157,9 @@ def train_batch(
     ) -> Dict[str, torch.Tensor]:
         input, target = batch
         model.drop_path_prob = (
-            self.hparams["drop_path_prob"] * (self.scheduler.last_epoch) / self.hparams["train_epochs"]
```
Comment (naren-determined) [inline examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py:152]: I actually kind of like how this was in one line before (seems more readable to me)

File: examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py [hunk 3]
```diff
@@ -165,18 +175,44 @@ def train_batch(
...
+        results = {
+            "loss": loss_avg.item() / num_batches,
+            "top1_accuracy": acc_top1.item() / num_batches,
+            "top5_accuracy": acc_top5.item() / num_batches,
+        }
+        if results["top1_accuracy"] > self.results["top1_accuracy"]:
+            self.results = results
 
-        return {"loss": loss, "top1_accuracy": top1, "top5_accuracy": top5}
+        return self.results
```
Comment (naren-determined) [inline examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py:211]: Wouldn't this make validation performance worse for dtrain since `evaluate_full_dataset` is not distributed?

Comment (liamcli) [reply examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py:211]: The goal here is to provide a better searcher metric for early stopping so that the signal of architecture accuracy is more stable not to validate faster.  Also, the architecture search phase is performed on CIFAR10 and would not benefit much from dtrain.  In the architecture evaluation phase, we do not use evaluate_full_dataset.

File: examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py (hunk 2 above)
Comment (liamcli) [reply examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py:152]: This was outputted by black formatting.  I think okay to keep.

File: examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py [hunk 4]
```diff
@@ -165,18 +175,44 @@ def train_batch(
...
-        logits, _ = model(input)
-        loss = self.criterion(logits, target)
-        top1, top5 = utils.accuracy(logits, target, topk=(1, 5))
+    # def evaluate_batch(self, batch: Any, model: nn.Module) -> Dict[str, Any]:
+    #    input, target = batch
+    #    logits, _ = model(input)
+    #    loss = self.criterion(logits, target)
+    #    top1, top5 = utils.accuracy(logits, target, topk=(1, 5))
+
+    #    return {"loss": loss, "top1_accuracy": top1, "top5_accuracy": top5}
```
Comment (naren-determined) [inline examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py:184]: I would rather remove this than keep it commented

**Review by naren-determined — APPROVED** [review]
I would prefer removing the commented `evaluate_batch` section and just keeping the `evaluate_full_dataset`, but otherwise looks good!
